sql/driver/tx.go

```diff
@@ -17,16 +17,56 @@
 
 package driver
 
-import "fmt"
+import "database/sql/driver"
 
 type tx struct {
 	conn *conn
+	txn  []byte
```

Comment:
The conn should be pinned to the transaction for the lifetime of the transaction (the Go SQL client handles this). I don't think this `txn` field is necessary.